crux-mir/src/Mir/Trans.hs

```diff
@@ -1525,6 +1525,10 @@ initialValue (CTyArray t) = tyToReprM t >>= \(Some tpr) -> case tpr of
 initialValue ty@(CTyBv _sz) = tyToReprM ty >>= \(Some tpr) -> case tpr of
     C.BVRepr w -> return $ Just $ MirExp (C.BVRepr w) $ S.app $ eBVLit w 0
     _ -> mirFail $ "Bv type " ++ show ty ++ " does not have BVRepr"
+-- `Any` values have no reasonable default.  Any default we provide might get
+-- muxed with actual non-default values, which will fail (unless the concrete
+-- type happens to match exactly).
+initialValue CTyAny = return Nothing
```

Comment:
I'm a little confused by this change.  At a first glance it seems like this should make no difference, as the pattern doesn't appear previously and the fallthrough case already returns `Nothing`.  Is this a pattern synonym that overlaps with something else?

Reply:
`CTyAny` is an pattern synonym for `TyAdt` with a specific name.  Without this case, it falls through to the `M.TyAdt` `Struct` case.  (`crucible::any::Any` is declared as a struct in Rust, but its crucible representation is overridden in crux-mir.)  The `Struct` case happens to produce a crucible `AnyType` value (crux-mir wraps all Rust ADTs in `AnyType`), but the value it produces isn't useful or meaningful, and causes the error described above.